lib/color_spinor_field.cpp

```diff
@@ -41,18 +41,21 @@ namespace quda {
     destroy();
   }
 
-  void ColorSpinorField::createGhostZone(int nFace, bool spin_project) const {
-
+  void ColorSpinorField::createGhostZone(int nFace, bool spin_project) const
+  {
     if ( typeid(*this) == typeid(cpuColorSpinorField) || ghost_precision_allocated == ghost_precision ) return;
 
-    // For Wilson we half the number of effective faces if the fields are spin projected.
-    int num_faces = ((nSpin == 4 && spin_project) ? 1 : 2) * nFace;
-    int num_norm_faces = 2*nFace;
+    bool is_fixed = (ghost_precision == QUDA_HALF_PRECISION || ghost_precision == QUDA_QUARTER_PRECISION);
+    int nSpinGhost = (nSpin == 4 && spin_project) ? 2 : nSpin;
+    size_t site_size = nSpinGhost * nColor * 2 * ghost_precision + (is_fixed ? sizeof(float) : 0);
```

Comment:
Does this need a factor of `nVec`? (Based on the changes in `include/color_spinor_field.h`)

Reply:
Discuss offline.  This isn't an issue.